app/src/main/java/com/teamwss/websoso/ui/postNovel/postNovelViewModel/PostNovelViewModel.kt

```diff
@@ -0,0 +1,204 @@
+
+    val isStartDateVisible: LiveData<Boolean> get() = _isStartDateVisible
+    private val _isEndDateVisible = MutableLiveData<Boolean>()
+    val isEndDateVisible: LiveData<Boolean> get() = _isEndDateVisible
+
+    private val _platforms = MutableLiveData<List<EditResponse.Platform>>()
+    val platforms: LiveData<List<EditResponse.Platform>> get() = _platforms
+    private val _naverUrl = MutableLiveData<String>()
+    val naverUrl: LiveData<String> get() = _naverUrl
+    private val _kakaoUrl = MutableLiveData<String>()
+    val kakaoUrl: LiveData<String> get() = _kakaoUrl
+
+    fun getUserNovelInfo() {
+        _editResponse.value = EditResponse(
+            novelId = 1,
+            userNovelTitle = "재혼황후",
+            userNovelAuthor = "알파타르트",
+            userNovelGenre = "궁중 로맨스, 후회, 계약, 재혼",
+            userNovelImg = "https://i.namu.wiki/i/Fk6vktj6y1nvSrlUst8PrRzPlPm1YDrPsUF2Goe4sLt1ZLeyMsblasD1QPJ85QJTSuit5F93ApG6R1wXZFSq-huKCJ4DrR18TN-hllzYacrHpGlPcPNfUv_QY0PLKs2ASfW09lTCOQl8TMfXxGvkyw.webp",
+            userNovelDescription = "완벽한 황후였다. 그러나 황제는 도움이 될 황후가 필요 없다고 한다. 그가 원하는 건 배우자이지 동료가 아니라 한다.  황제는 나비에를 버리고 노예 출신의 여자를 옆에 두었다. 그래도 괜찮았다. 황제가 그녀에게 다음 황후 자리를 약속하는 걸 듣기 전까진.  나비에는 고민 끝에 결심했다. 그렇다면 난 옆 나라의 황제와 재혼하겠다고.",
+            userNovelRating = 4.5f,
+            userNovelReadStatus = "FINISH",
+            readStartDate = "2023-06-30",
+            readEndDate = "2024-01-11",
+            platforms = listOf(
+                EditResponse.Platform(
+                    "네이버시리즈",
+                    "https://series.naver.com/novel/detail.series?productNo=3713078"
+                ),
+            ),
+        )
+    }
+
+    fun updateReadStatus(readStatus: String) {
+        _readStatus.value = readStatus
+    }
+
+    fun updateReadDate(startDate: String, endDate: String) {
+        _startDate.value = startDate
+        _endDate.value = endDate
+    }
+
+    fun updateSelectedDate(selectedStartDate: String, selectedEndDate: String) {
+        _selectedStartDate.value = selectedStartDate
+        _selectedEndDate.value = selectedEndDate
+    }
+
+    fun updateRating(novelRating: Float) {
+        _rating.value = novelRating
+    }
+
+    fun updateIsDialogShown(isShown: Boolean = false) {
+        if (isShown) _isDialogShown.value = 0
+        else _isDialogShown.value = 8
+    }
+
+    fun updateIsNumberPickerStartSelected(isSelected: Boolean = true) {
+        _isNumberPickerStartSelected.value = isSelected
+    }
+
+    private fun splitDateToLocalDate(date: String): LocalDate {
+        return LocalDate.of(
+            date.split("-")[0].toInt(),
+            date.split("-")[1].toInt(),
+            date.split("-")[2].toInt()
+        )
+    }
+
+    fun formatToYear(date: String): Int {
+        return date.split("-")[0].toInt()
+    }
+
+    fun formatToMonth(date: String): Int {
+        return date.split("-")[1].toInt()
+    }
+
+    fun formatToDay(date: String): Int {
+        return date.split("-")[2].toInt()
+    }
```

Comment:
함수들의 중복되는 것 같은데 다른 방법으로 더 깔끔하게 못 작성할까요 ?
```suggestion
    private fun splitDateToLocalDate(date: String): LocalDate {
        return LocalDate.of(
        formatToYear(date),
           formatToMonth(date),
            formatToDay(date)
        )
    }

    fun formatToYear(date: String): Int {
        return date.split("-")[0].toInt()
    }

    fun formatToMonth(date: String): Int {
        return date.split("-")[1].toInt()
    }

    fun formatToDay(date: String): Int {
        return date.split("-")[2].toInt()
    }
```
이런식으로 ?

Reply:
아 이부분은 함수 개선 가능할 것 같군요 술깨고 진행하겠습니당

Reply:
good